packages/ics721/src/execute.rs

```diff
@@ -34,17 +41,29 @@ where
         msg: InstantiateMsg,
     ) -> StdResult<Response<T>> {
         CW721_CODE_ID.save(deps.storage, &msg.cw721_base_code_id)?;
-        PROXY.save(deps.storage, &None)?;
+        // proxy contracts are optional
+        INCOMING_PROXY.save(deps.storage, &None)?;
+        OUTGOING_PROXY.save(deps.storage, &None)?;
         PO.set_pauser(deps.storage, deps.api, msg.pauser.as_deref())?;
 
-        let proxy_instantiate = msg
-            .proxy
-            .map(|m| m.into_wasm_msg(env.contract.address))
-            .map(|wasm| SubMsg::reply_on_success(wasm, INSTANTIATE_PROXY_REPLY_ID))
-            .map_or(vec![], |s| vec![s]);
+        let mut proxies_instantiate: Vec<SubMsg<T>> = Vec::new();
+        if let Some(cii) = msg.incoming_proxy {
```

Comment:
What is the benefit of doing this other than convenience?

Reply:
This code exists since a while. Besides this: it is an hen-egg problem: if you dont instantiate proxy, whilie instantiating ics721, then you need to:
- instantiate proxy manually (and set ics721 as `origin`)
- migrate ics721 and set proxy

This code here requires only one more step:
- migrate proxy and set ics721 as proxy

Reply:
Yeah so you're trying to save a migration step. I'm not sure which way is worse - having to do additional administration steps or code that is doing too much. It doesn't seem like there's a clear answer, but we should be aware of how complicated ics721 is getting in general with incoming proxies, outgoing proxies, circuit breakers, withdraw addresses etc.